### PR TITLE
Document Convention and Topology

### PR DESCRIPTION
**Pull request by morty29**

### CODE REVIEW

File: connection/request_executor.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package connection
+
+type RequestExecutor struct{
+
+	url string
+	database string
+	apiKey string
+	convenion string
```
Comment (ayende) [inline connection/request_executor.go]: What is the meaning of this field?

Comment (ayende) [inline connection/request_executor.go]: Also, typo

File: d_commands/commands.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+	//Specifies if only document metadata should be returned
+	metadataOnly bool
+}
+
+type DeleteDocumentCommand struct{
+
+	ravenCommand RavenCommand
+
+	//unique key under which document will be stored
```
Comment (ayende) [inline d_commands/commands.go]: The comment is wrong, this is the key to be _deleted_

File: d_commands/commands.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	key string
+
+	//current document etag, used for concurrency checks (nil to skip check)
+	etag int
```
Comment (ayende) [inline d_commands/commands.go]: etags are `int64` not `int`.

Comment (ayende) [inline d_commands/commands.go]: How do you indicate a null etag (don't do the check)?

File: d_commands/commands.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+	ravenCommand RavenCommand
+
+	//unique key under which document will be stored
+	key string
+
+	//document data
+	document []interface{}
+
+	//current document etag, used for concurrency checks (nil to skip check)
+	etag int
```
Comment (ayende) [inline d_commands/commands.go]: Same etag issue.

File: data/document_convention.go [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+type WriteBehaviour struct {
+	behaviour Behaviour
+}
+
+type DocumentConvention struct {
+	MaxNumberOfRequestsPerSession, MaxIdsToCatch,
+	Timeout, MaxLengthOfQueryUsingGetUrl uint
+	DefaultUseOptimisticConcurrency bool
+	systemDB, IdentityPartsSeparator string
```
Comment (ayende) [inline data/document_convention.go]: Why is the casing of `systemDB` different?
For that matter, what is it supposed to do?

**Comment by igor-chyhrynets** [comment]
It indicates that the member of the module is private

File: data/document_convention.go (hunk 5 above)
Comment (igor-chyhrynets) [inline data/document_convention.go]: It indicates that the member of the module is private

File: connection/request_executor.go (hunk 1 above)
Comment (igor-chyhrynets) [inline connection/request_executor.go]: Yea, type was a placeholder, will change to proper one.

File: data/document_convention.go (hunk 5 above)
Comment (ayende) [inline data/document_convention.go]: What what is the purpose of this field?

Comment (igor-chyhrynets) [inline data/document_convention.go]: Yeah, it is not used anywhere despite its presence in python client. I took it without paying attention to it. Will remove.

**Pull request merged**